synchronizer/l1_live_block_ranges.go

```diff
@@ -59,18 +66,43 @@ func (b *blockRange) overlaps(br blockRange) bool {
 	return b.fromBlock <= br.toBlock && br.fromBlock <= b.toBlock
 }
 
-func (l *liveBlockRanges) addBlockRange(br blockRange) error {
+func (l *liveBlockRangesGeneric[T]) addBlockRange(br blockRange) error {
+	var zeroValue T
+	return l.addBlockRangeWithTag(br, zeroValue)
+}
+
+func (l *liveBlockRangesGeneric[T]) addBlockRangeWithTag(br blockRange, tag T) error {
 	if err := br.isValid(); err != nil {
 		return err
 	}
 	if l.overlaps(br) {
 		return errBlockRangeInvalidOverlap
 	}
-	l.ranges = append(l.ranges, liveBlockRangeItem{br})
+	l.ranges = append(l.ranges, liveBlockRangeItem[T]{blockRange: br, tag: tag})
 	return nil
 }
 
-func (l *liveBlockRanges) removeBlockRange(br blockRange) error {
+func (l *liveBlockRangesGeneric[T]) setTagByBlockRange(br blockRange, tag T) error {
+	for i, r := range l.ranges {
+		if r.blockRange == br {
+			l.ranges[i].tag = tag
+			return nil
+		}
+	}
+	return errBlockRangeNotFound
+}
```

Comment:
Why don't you set the tag in func addBlockRange ?

Reply:
It's done in that way: first time it call to `addBlockRangeWithTag` that add the range with the tag. 
Later, it's call  `setTagByBlockRange` to update the last time that we try this range